File: api/Bookshop.Application/Features/Customers/CustomerResponseDto.cs
﻿using Bookshop.Application.Features.ShoppingCarts;

namespace Bookshop.Application.Features.Customers
{
    public class CustomerResponseDto
    {
        public long Id { get; set; }
        public string? UserName { get; set; }
        public string? Email { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? ShippingAddress { get; set; }
        public string? BillingAddress { get; set; }
        public ShoppingCartResponseDto? ShoppingCart {  get; set; }
    }
}

File: ui/src/app/dto/customer/customer-response-dto.ts
import { ShoppingCartResponseDto } from "../shoppingcart/shoppingcart-response-dto";

export class CustomerResponseDto {
  userName: string = '';
  email: string = '';
  firstName: string = '';
  lastName: string = '';
  shippingAddress: string = '';
  billingAddress: string = '';
  shoppingCart: ShoppingCartResponseDto | undefined = undefined;

  constructor(data?: Partial<CustomerResponseDto>) {
    this.userName = data?.userName ?? this.userName;
    this.email = data?.email ?? this.email;
    this.firstName = data?.firstName ?? this.firstName;
    this.lastName = data?.lastName ?? this.lastName;
    this.shippingAddress = data?.shippingAddress ?? this.shippingAddress;
    this.billingAddress = data?.billingAddress ?? this.billingAddress;
    this.shoppingCart = new ShoppingCartResponseDto(data?.shoppingCart) ?? this.shoppingCart;
  }
}